code/controllers/subsystem/evacuation.dm

```diff
@@ -0,0 +1,240 @@
+
+/datum/controller/subsystem/evacuation/proc/remove_evacuation_blocker(datum/bad)
+	evacuation_blockers -= bad
+	if(!length(evacuation_blockers))
+		for(var/identifier in controllers)
+			controllers[identifier].evacuation_unblocked()
+
+/datum/controller/subsystem/evacuation/proc/disable_evacuation(controller_id)
+	if(!controllers[controller_id])
+		CRASH("Tried to disable evacuation of an invalid controller ID")
+	controllers[controller_id].disable_evacuation()
+
+/datum/controller/subsystem/evacuation/proc/enable_evacuation(controller_id)
+	if(!controllers[controller_id])
+		CRASH("Tried to enable evacuation of an invalid controller ID")
+	controllers[controller_id].enable_evacuation()
+
+/datum/controller/subsystem/evacuation/proc/block_cancel(controller_id)
+	if(!controllers[controller_id])
+		CRASH("Tried to block cancel of an invalid controller ID")
+	controllers[controller_id].block_cancel()
+
+/datum/controller/subsystem/evacuation/proc/unblock_cancel(controller_id)
+	if(!controllers[controller_id])
+		CRASH("Tried to unblock cancel of an invalid controller ID")
+	controllers[controller_id].unblock_cancel()
+
+//Perhaps move it to SShuttle?
+/datum/controller/subsystem/evacuation/proc/get_customizable_shuttles()
+	var/list/shuttles = list()
+	for(var/identifier in controllers)
+		shuttles += controllers[identifier].get_customizable_shuttles()
+	return shuttles
+
+/datum/controller/subsystem/evacuation/proc/get_endgame_areas()
+	var/list/areas = list()
+	for(var/identifier in controllers)
+		var/datum/evacuation_controller/controller = controllers[identifier]
+		// We only want to add the areas if the controller is in a state where evacuation has finished
+		if(controller.state >= EVACUATION_STATE_EVACUATED)
+			areas += controller.get_endgame_areas()
+	return areas
+
+/datum/controller/subsystem/evacuation/proc/get_stat_data()
+	var/list/data = list()
+	for(var/identifier in controllers)
+		data += controllers[identifier].get_stat_data()
+	return data
+
+/datum/controller/subsystem/evacuation/proc/get_world_topic_status()
+	var/list/status = list()
+	for(var/identifier in controllers)
+		status += controllers[identifier].get_world_topic_status()
+	return status
+
+/datum/controller/subsystem/evacuation/proc/evacuation_in_progress()
+	for(var/identifier in controllers)
+		if(controllers[identifier].state == EVACUATION_STATE_AWAITING)
+			return TRUE
+	return FALSE
+
+/datum/controller/subsystem/evacuation/proc/evacuation_can_be_cancelled()
+	for(var/identifier in controllers)
+		if(!controllers[identifier].can_cancel(null))
+			return FALSE
+	return TRUE
+
+/datum/controller/subsystem/evacuation/proc/station_evacuated()
+	for(var/identifier in controllers)
+		if(controllers[identifier].state >= EVACUATION_STATE_EVACUATED)
+			return TRUE
+	return FALSE
+
+/datum/controller/subsystem/evacuation/proc/delay_evacuation(identifier, delay)
+	if(!controllers[identifier])
+		CRASH("Tried to delay evacuation of an invalid controller ID")
```

Comment:
```suggestion
		CRASH("Tried to delay evacuation of an invalid controller ID: [identifier]")
```

Also, why does this not follow the standard argument naming convention of `controller_id`?